OSDOCS-5275: Updated VSphere platform specification (#55767)
openshift/openshift-docs: modules/installation-configuration-parameters.adoc

|====
|Parameter|Description|Values

|`platform.vsphere`

> | Parameter covers two purposes in the context of a vSphere configuration: (1) Describes your account on the cloud platform that hosts your cluster, and you use the parameter to customize the platform; (2) The parameter is optional when providing additional configuration settings for the machine pool parameters associated with compute and control plane machines.

Author: Nice suggestion, updated. Yes, the lead-in sentence is the mandatory entry. The optional clause is for a specific configuration.

a|String
[NOTE]
====
You can only specify one vCenter server for your {product-title} cluster.
====

> |`platform.vsphere.vCenters`

Reviewer: platform.vsphere.vCenters is not a valid field
```
sh-4.4$  ./openshift-install explain installconfig.platform.vCenters
FATAL Error executing openshift-install: failed to load schema for the field platform.vCenters: invalid field vCenters, no such property found
```
see 
```
sh-4.4$  ./openshift-install explain installconfig.platform.vsphere.vcenters
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <[]object>
  VCenters holds the connection details for services to communicate with vCenter. Currently only a single vCenter is supported.

FIELDS:
    datacenters <[]string> -required-
      Datacenter in which VMs are located.

    password <string> -required-
      Password is the password for the user to use to connect to the vCenter.

    port <integer>
      Default: 443
      Format: int32
      port is the TCP port that will be used to communicate to the vCenter endpoint. This is typically unchanged from the default of HTTPS TCP/443.

    server <string> -required-
      server is the fully-qualified domain name or the IP address of the vCenter server.

    user <string> -required-
      Username is the username that will be used to connect to vCenter
```

Author: Updated

|Lists any fully-qualified hostname or IP address of a vCenter server.
|String

> |`platform.vsphere.datacenters`

Reviewer: platform.vsphere.datacenters also not a valid field, seem like it should be platform.vsphere.vcenters.datacenters, but should we also list other fields under vcenters here?
```
sh-4.4$  ./openshift-install explain installconfig.platform.vsphere.datacenters
FATAL Error executing openshift-install: failed to load schema for the field platform.vsphere.datacenters: invalid field datacenters, no such property found
```

Author: Updated.  @rvanderp3 , WDYT of the comment `but should we also list other fields under vcenters here?` ?

Reviewer 2: yep, good catch @sgaoshang , we should list datacenters as well under vcenters

Author: Sorry, @sgaoshang and @rvanderp3 . What do I need to list in the table regarding `installconfig.platform.vsphere.datacenter`?

|Lists and defines the datacenters where {product-title} virtual machines (VMs) operate. The list of datacenters must match the list of datacenters specified in the `failureDomains` field.
|String

|`platform.vsphere.apiVIPs`
|Virtual IP (VIP) addresses that you configured for control plane API access.
a|Multiple IP addresses

[NOTE]
====
In {product-title} 4.12 and later, the `apiVIP` configuration setting is deprecated. Instead, use a list format to enter a value in the `apiVIPs` configuration setting.

> ====

Author: Moved.

|`platform.vsphere.failureDomains`
|Establishes the relationships between a region and zone. You define a failure domain by using vCenter objects, such as a `datastore` object. A failure domain defines the vCenter location for {product-title} cluster nodes.
|String

> |`platform.vsphere.region`

Reviewer: it should under platform.vsphere.failureDomains.region

Author: Updated

|You define a region by using a tag from the `openshift-region` tag category. The tag must be attached to the vCenter datacenter.
|String

> |`platform.vsphere.zone`

Reviewer: it should under platform.vsphere.failureDomains.zone

Author: Updated

|You define a zone by using a tag from the `openshift-zone` tag category. The tag must be attached to the vCenter datacenter.
|`String`

> |`platform.vsphere.networks`

Reviewer: also only platform.vsphere.failureDomains.topology.networks exist

Author: Updated

|Lists any network in the vCenter instance that contains the virtual IP addresses and DNS records that you configured.
|String

|`platform.vsphere.ingressVIPs`
|Virtual IP (VIP) addresses that you configured for cluster Ingress.
a|Multiple IP addresses

[NOTE]
====
In {product-title} 4.12 and later, the `ingressVIP` configuration setting is deprecated. Instead, use a list format to enter a value in the `ingressVIPs` configuration setting.

> ====

Author: Moved.

|`platform.vsphere.diskType`
|Optional. The disk provisioning method. This value defaults to the vSphere default storage policy if not set.
|Valid values are `thin`, `thick`, or `eagerZeroedThick`.
|====

[id="deprecated-parameters-vsphere_{context}"]
== Deprecated VMware vSphere configuration parameters

> The {product-title} 4.13 deprecates some vSphere configuration parameters. You can continue to use these parameters, but the installation program does not automatically specify these parameters in the `install-config.yaml` file

Reviewer 3: ```suggestion
In {product-title} 4.13, the following vSphere configuration parameters are deprecated. You can continue to use these parameters, but the installation program does not automatically specify these parameters in the `install-config.yaml` file.
```

|Integer
|====

[id="installation-configuration-parameters-zone-region-vsphere_{context}"]
== VMware vSphere zone and region enablement configuration parameters

The following table describes VMware vSphere zone and region enablement configuration parameters:

.VMware vSphere zone and region enablement parameters
[cols=".^2,.^3,.^3a",options="header"]
|====
|Parameter|Description|Values

|`vsphere`
|Parameter covers two purposes in the context of a vSphere configuration: (1) Describes your account on the cloud platform that hosts your cluster, and you use the parameter to customize the platform; (2) The parameter is optional when providing additional configuration settings for the machine pool parameters associated with compute and control plane machines.

a|String

[NOTE]
====
You can only specify one vCenter server for your {product-title} cluster.
====

|`vcenters`
| The fully-qualified hostname or IP address for each listed vCenter server.
| String

|`datacenters`

> |Defines the multiple datacenters that operate in a single VMware vCenter.

Reviewer 4: Something like this instead:
Defines the datacenters where the OCP virtual machines will operate. That list of datacenters **must** match the list of datacenters used in failuredomains.

Author: Thank you. I updated with some slight changes.

|String

|`failureDomains`
|Establishes the relationships between a region and zone. You define a failure domain by using vCenter objects, such as a `datastore` object. A failure domain defines the vCenter location for {product-title} cluster nodes.
|String

|`region`

> |Specifies a vCenter datacenter. You define a region by using a tag from a created tag category, such as `openshift-region`.

Reviewer 4: ```suggestion
|You define a region by using a tag from the created tag category `openshift-region`. This tag is attached to a vCenter datacenter.
```

|String

|`zone`

> |Specifies a vCenter cluster. You define a region by using a tag from from a created tag category, such as `openshift-zone`.

Author: Updated

|`String`

|`folder`

> |Optional parameter that stores manifest configuration files for the cluster.

Reviewer 4: ```suggestion
|Optional parameter that defines a virtual machine folder within in vcenter that the RHCOS virtual machines will be provisioned into.
```

Author: :heavy_check_mark:

|Integer
|====

> [id="installation-configuration-parameters-zone-region-vsphere_{context}"]

Author: Amalgamated the parameters into one table. Worth keeping the deprecated parameters?

See https://github.com/openshift/installer/blob/master/pkg/types/vsphere/platform.go#L35-L128
